Adding theming documentation (#9)
gr4vy/gr4vy-docs: guides/embed/theming.mdx

## Usage

To apply a theme, embed can be passed an Object containing [Theme Options](#theme-options).

**The theme options will override the default theme**.

<Tabs
  groupId="language"
  defaultValue="react"
  values={[
    { label: 'React', value: 'react', },
    { label: 'CDN', value: 'cdn', },
  ]}
>
  <TabItem value="react">

```tsx
<Gr4vyEmbed
  currency="GBP"
  amount="2300"
  theme={{
    colors: {
      primary: "green",
      text: "black",
    },
    borderWidths: {
      input: "thick",
    },
    radii: {
      input: "none",
    },
  }}
/>
```

  </TabItem>

  <TabItem value="cdn">

```js
setup({
  gr4vyId: "acme",
  element: ".container",
  amount: 1299,
  country='US'
  currency: "USD",
  token: token,
  theme: {
    colors: {
      primary: "green",
      text: "black",
    },
    borderWidths: {
      input: "thick",
    },
    radii: {
      input: "none",
    },
  }
});
/>
```

  </TabItem>
</Tabs>

## Theme Options

### Fonts

Fonts can be customized with a valid CSS property.

```js
{
  fonts: {
    /**
     * Body - Used for all text
     */
    body: "google:Lato, Tahoma, Arial";
  }
}
```

:::info Using Google Fonts

> Fonts prefixed with `google:` are downloaded automatically from [Google Fonts](https://fonts.google.com/). This allows you to choose the closest font to your brand. We recommend choosing a font with weights: `400`, `600` and `700`.

Reviewer: ```suggestion
Fonts prefixed with `google:` are downloaded automatically from [Google Fonts](https://fonts.google.com/). This allows you to choose the closest font to your brand. We recommend choosing a font with support for the weights `400`, `600` and `700`.
```

:::

### Colors

> Colors are named semanticly e.g. `primary` or `danger`. This means we can theme new features or UI elements without explicitly asking you to update colors. As an example, we may introdce a 'Remove Card' button at a later stage that would be based of the `danger` set of colors.

Reviewer: ```suggestion
Colors are named semanticly, for example `primary` or `danger`. This allows us to update introduce new features or UI elements without explicitly requiring you to update your theme. As an example, we may introduce a 'Remove Card' button at a later stage that would be based on the `danger` set of colors.
```

     */
    input: 'thin' // 'thin' or 'thick'
  }
}
```

### Radii

Radii can be set to either `none`, `subtle` or `rounded`. These aliases correspond to `0`, `2px` and `4px`.
Setting the radii to `none` will remove it entirely.

```js
{
  radii: {
    /**
     * Container - A bounding layout element
     */
    container: "subtle"; // 'subtle', 'rounded' or 'none'
    /**
     * Input - A Form Input, e.g. Card Number
     */
    input: "subtle"; // 'subtle', 'rounded' or 'none'
  }
}
```

### Focus Ring

The focus ring is set using a CSS `box-shadow` instead of the default browser focus ring. It can be overriden, but we advise at least keeping a visible focus ring for accessibility.
This will override any focus styles set elsewhere in the theme.

```js
{
  shadows: {
    /**
     * Focus Ring - Surrounds an element that is currently focused
     */
    focusRing: "0 0 0 2px #ffffff, 0 0 0 4px #4844ff";
  }
}
```

:::info Reference an existing theme option

Existing theme options can be referenced using CSS variables. This object will be flattened to CSS variables, e.g. `colors.primary` will become `--colors-primary` which
can be used as follows: `0 0 0 3px (--colors-primary)`.

> :::

Reviewer: Do we really want to document this? Does this lock us in to using CSS variables?

Author: I can remove this - It does lock us to this format, you're right. e.g. I can see later on when we move some styles to the merchant page, we will need to namespace these with `--gr4vy-colors-primary`, so probably best to leave this out for the time being.

Reviewer: Yeah let's remove.

## FAQ

> ### How do I add custom styles?

Reviewer: ```suggestion
### How do I load my own CSS files?
```

Suggestion as I currently think our theme allows you to set your own style.

We currently do not support adding your own styles directly via CSS.

> This allows us to roll out updates and ensure compatibility with new features.

Reviewer: ```suggestion
Using a theme allows us to roll out updates and ensure compatibility with new features.
```

### What browsers are supported?

> Theming uses CSS variables under-the-hood and so for browsers that do not support CSS variables (e.g. IE11) the default theme will be used.

Reviewer: ```suggestion
Theming is currently supported in all modern browsers including Microsoft Edge. Some browsers like Internet Explorer 11 are not supported, and in these the default theme will be used.
```

### What values can be set?

> Values can be a valid CSS property value. e.g. `colors.primary` could be a hex, rgb or hsl. You can also have multiple shadows for a focus ring, e.g. `0 0 0 2px white, 0 0 0 4px green`

Reviewer: Remind me? Do we do anything to stop a user from loading any `url`s?

Author: Yep - There is a strict Content Security Policy, which should only allow requests to their Gr4vy cluster and google fonts.

```
"default-src 'self' https://*.gr4vy.app; img-src https://*.gr4vy.app data:; style-src 'self' https://fonts.googleapis.com; font-src https://fonts.gstatic.com;"
```

This should be enough to mitigate IMO, but we could do some further sanitzation, or stripping of values.
